marimo-tutorials/Data-Science/Exploratory-Data-Analysis/Goodreads-EDA.py

```diff
@@ -0,0 +1,1271 @@
+        - Max - 5, Min - 1
+        ### 5.  How many total columns are there in dataset books_enriched.csv
+        - 30
+        """
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def __(mo):
+    mo.md(
+        r"""
+        ### 6.	Anything noticeable in books dataset? If Yes,what? What steps will you suggest to handle that
+        ### 7.	Print the statistical summary of books dataset and write the inference from the statistics observed for numeric as well as categorical columns.
+        - From the above, while using the info() and describe() commands, I noticed the following:
+          1. There are two columns for authors (authors and authors_2).
+          2. There exist certain values in the publication year (-1750)
+          3. No pages available for the book to actually exist (and yet it shows as published).
+          4. Certain columns and attributes are not relevant and can be dropped during the PCA, EDA and feature engineering process to utilize data for various use case scenarios.
+          5. For ratings_count value, we notice that the maximum value is 4780653 ratings while the minimum value is 2716 ratings.
+        """
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def __(df2, mo):
+    with mo.redirect_stdout():
+        # Statistical summary for numeric columns
+        numeric_summary = df2.describe()
+
+        # Statistical summary for categorical columns
+        categorical_summary = df2.describe(include=["object"])
+
+        print("Statistical Summary for Numeric Columns:")
+        print(numeric_summary)
+
+        print("\nStatistical Summary for Categorical Columns:")
+        print(categorical_summary)
+
+        # Inferences
+        numeric_inferences = """
+        Numeric Columns Summary:
+        1. average_rating: Mean rating is around 4.0 indicating a generally positive user rating for books.
+        2. books_count: This varies widely, with some books having multiple editions (max is 4917).
+        3. original_publication_year: The mean year is around 1982, indicating the dataset has a mix of both old and recent books.
+        4. pages: Mean pages per book is around 336, but this varies greatly with some very short and very long books.
+        """
+
+        categorical_inferences = """
+        Categorical Columns Summary:
+        1. authors: Multiple authors are common; the dataset has unique author names for many books.
+        2. genres: Books are classified into multiple genres, indicating diverse book content.
+        3. language_code: 'eng' is the most common language code, suggesting the majority of books are in English.
+        4. title: Each book has a unique title, though some titles may be shared by different works.
+        """
+
+        print(numeric_inferences)
+        print(categorical_inferences)
+    return (
+        categorical_inferences,
+        categorical_summary,
+        numeric_inferences,
+        numeric_summary,
+    )
+
```

Comment:
Instead of printing, maybe using marimo table?
```python
mo.vstack(
    [
        mo.md("#### Statistical summary for numeric columns"),
        numeric_summary,
        mo.md("#### Statistical Summary for Categorical Columns"),
        categorical_summary
    ]
)
```